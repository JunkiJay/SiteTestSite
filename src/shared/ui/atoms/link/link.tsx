import { forwardRef, MouseEventHandler } from 'react';
import { Link as RouterLink, LinkProps as RouterLinkProps } from 'react-router-dom';
import { styled } from '@mui/material/styles';
import { useHapticFeedback } from '@tma.js/sdk-react';

interface LinkProps extends RouterLinkProps {}

const StyledRouterLink = styled(RouterLink)<LinkProps>``;
export const Link = forwardRef<HTMLAnchorElement, LinkProps>(({ onClick, ...props }, ref) => {
  const haptic = useHapticFeedback();

  const handleClick: MouseEventHandler<HTMLAnchorElement> = (e) => {
    haptic.impactOccurred('medium');

    if (onClick) {
      onClick(e);
    }
  };

  return <StyledRouterLink ref={ref} onClick={handleClick} {...props} />;
});
