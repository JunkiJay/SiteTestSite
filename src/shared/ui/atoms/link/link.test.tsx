import { render, screen } from '@testing-library/react';
import { Link } from './link';
import { MemoryRouter } from 'react-router-dom';

describe('Link', () => {
  it('Match Snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Link to="#">Test</Link>
      </MemoryRouter>,
    );

    expect(screen.getByText('Test')).toHaveTextContent('Test');

    expect(container).toMatchSnapshot();
  });

  it('Rendered', () => {
    render(
      <MemoryRouter>
        <Link to="#">Test</Link>
      </MemoryRouter>,
    );

    const element = screen.getByText('Test');

    expect(element).toBeInTheDocument();
  });
});
