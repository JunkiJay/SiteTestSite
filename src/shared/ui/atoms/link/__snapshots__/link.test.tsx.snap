// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Match Snapshot 1`] = `
<div>
  <a
    class="css-0"
    href="/"
  >
    Test
  </a>
</div>
`;

exports[`Rendered 1`] = `
<div>
  <a
    class="css-0"
    href="/"
  >
    Test
  </a>
</div>
`;
